src/improvements/tasks/MultisigTask.sol

```diff
@@ -349,17 +353,15 @@ abstract contract MultisigTask is Test, Script, StateOverrideManager, TaskManage
             );
         }
 
-        require(IGnosisSafe(parentMultisig).nonce() == nonce + 1, "MultisigTask: nonce not incremented");
+        require(IGnosisSafe(rootSafe).nonce() == originalRootSafeNonce + 1, "MultisigTask: nonce not incremented");
 
-        _validate(accountAccesses, actions);
+        _validate(accountAccesses, actions, rootSafe);
 
         checkStateDiff(accountAccesses);
     }
 
     /// @notice Get the multicall address for the given safe. Override to return required multicall address.
-    function _getMulticallAddress(address safe) internal view virtual returns (address) {
-        // Some child contracts may override this function and return a different multicall address
-        // based on the safe address (e.g. whether it's the parent or child multisig).
+    function _getMulticallAddress(address safe, address[] memory) internal view virtual returns (address) {
```

Comment:
takes `allSafes` as an arg so that child contracts can determine if `safe` is root or not.